src/main/java/net/snowflake/client/core/SessionUtil.java

```diff
@@ -1006,6 +1020,12 @@ static void closeSession(SFLoginInput loginInput) throws SFException, SnowflakeS
 
       postRequest = new HttpPost(uriBuilder.build());
 
+      // Add custom headers before adding common headers
+      Map<String, String> additionalHttpHeaders = loginInput.getAdditionalHttpHeaders();
```

Comment:
As this block used many places create a util method and call.

Reply:
Moved them all to a util, and the util is now checking for not null and not empty.